lib/business_logic/bloc/to_do_list/to_do_list_bloc.dart

```diff
@@ -0,0 +1,32 @@
+import 'dart:async';
+
+import 'package:bloc/bloc.dart';
+import 'package:flutter_boilerplate/business_logic/services/api_calls/todo_api.dart';
+import 'package:flutter_boilerplate/business_logic/services/models/todolist.dart';
+import 'package:rxdart/rxdart.dart';
+import 'package:meta/meta.dart';
+
+part 'to_do_list_event.dart';
+part 'to_do_list_state.dart';
+
+class ToDoListBloc extends Bloc<ToDoListEvent, ToDoListState> {
+  TodoApi todo = new TodoApi();
+  ToDoListBloc() : super(ToDoListInitial());
+  final _todoFetcher = PublishSubject<List<TodoList>>();
+
+  Stream<List<TodoList>> get todoListStream => _todoFetcher.stream;
```

Comment:
We should use this inside the other two files (the event and the state) because we should control the state from there. You can take this example: https://itnext.io/bloc-pattern-in-flutter-explained-with-real-example-a858e69eb5b8, also you can view the other tiny example that we have here in the baseline with the bloc.